core/src/main/scala/com/criteo/cuttle/platforms/local/package.scala

```diff
@@ -2,17 +2,54 @@ package com.criteo.cuttle.platforms
 
 /** Allow to fork process locally in a managed way.
   *
-  * It provides the `sh` interpolator that you can use to fork shell scripts:
+  * It provides the `exec` interpolator that you can use to fork shell commands:
   *
   * {{{
-  *   sh"""hdfs dfs -cp /from /to""".exec()
+  *   exec"""hdfs dfs -cp /from /to"""()
   * }}}
+  *
+  * Your command(s) will be executed in a forked `sh -c exec $command` process.
+  * We can provide a guarantee that forked process will be killed when you want
+  * to cancel or pause it only when you put a single command into this interpolation.
+  * If you use multiple commands you should provide a way to kill all potentially forked
+  * children of your process.
+  *
+  * Actually there are two ways of chaining several commands that are semantically different:
+  *
+  * 1. Chain multiple execs in a for-comprehension. Because of platform mechanism
+  * it will require several platform allocations to run completely. And we can guarantee
+  * that forked process will be killed.
+  *
+  * Example:
+  *
+  * {{{
+  *   for {
+  *          _ <- exec"echo Hello"()
+  *          _ <- exec"echo Check my project page at https://github.com/criteo/cuttle"()
+  *          completed <- exec"sleep 1"()
+  *       }
+  * }}}
+  *
+  * 2. Wrap multiple commands in one ```sh -c```. _In this case we cannot guarantee that
+  * all process spawned inside this fork will be successfully killed when we kill the parent process._
+  *
+  * Example:
+  *
+  * {{{
+  *   exec"""sh -c '
+  *       |    echo 1
+  *       |    echo 2
+  *       |    # we fork a process here that will survive
+  *       |    sh -c "sleep 10000"
+  *       |'"""()
+  * }}}
```

Comment:
I think it is better to say something like:


> The command your provide will be forked into another process. Note that if you provide several commands separated by `;' only the first one will be forked and the other ones will be ignored.
> 
> If you really need to run several commands, you can use a shell script (for example using sh -ce '...' etc.).
> 
> Also, note that in case of job cancellation, Cuttle will kill the forked process, but if your process fork other processes (typically the case for a shell script) it is your responsibility to kill them in cascade.